[docs] Add a section about local apps (#31745)
expo/expo: docs/pages/workflow/overview.mdx

If you are using EAS Build, we recommend going through [Internal distribution](/build/internal-distribution/) to learn more about sharing your app for testing.

> If you compile your app locally, you can create [release builds locally](/guides/local-app-production/).

Author: ```suggestion
If you compile your app locally, you can create [production builds locally](/guides/local-app-production/).
```

## Release app to stores

To release your app on the app stores, you can use [EAS Submit](/submit/introduction/). For more information on using EAS Submit, see [Submit to Google Play Store](/submit/android/) and [Submit to Apple App Store](/submit/ios/).

> To create a release build locally, see [release builds locally](/guides/local-app-production/) and then go through the app stores guide to submit your app.

Author: ```suggestion
To create a production build locally, see [release builds locally](/guides/local-app-production/) and then go through the app stores guide to submit your app.
```

> To create a production build locally, see [release builds locally](/guides/local-app-production/) and then go through the app stores guide to submit your app.

Author: ```suggestion
To create a production build locally, see the [guide](/guides/local-app-production/) on the same and then go through the app stores guide to submit your app.
```
